src/Ipac_Table_Writer/internals.cxx

```diff
@@ -92,6 +151,31 @@ void tablator::Ipac_Table_Writer::write_consecutive_records(
 
 /**********************************************************/
 
+void tablator::Ipac_Table_Writer::write_consecutive_records(
+        const Table& table, std::ostream& os,
+        const std::vector<size_t>& included_column_ids, size_t start_row,
+        size_t row_count, const std::vector<Data_Type>& datatypes_for_writing) {
+    size_t num_table_rows = table.num_rows();
+    if (start_row > num_table_rows) {
+        std::string msg = "Invalid start_row ";
+        msg.append(std::to_string(start_row))
+                .append("; only")
+                .append(std::to_string(num_table_rows))
+                .append(" rows in table.");
+        throw std::runtime_error(msg);
+    }
+    size_t end_row = std::min(num_table_rows, start_row + row_count);
```

Comment:
`end_row = start_row + std::min(row_count, num_table_rows - start_row)` to avoid problems with integer overflow?